test/lib/pipeline.test.js

```diff
@@ -1037,6 +1037,56 @@ describe('Pipeline Model', () => {
             // as the model's pipeline property, now
             assert.calledOnce(secretFactoryMock.list);
         });
+
+        it('gets config pipeline\'s secrets', () => {
+            const childPipelineId = 1234;
+
+            pipelineConfig.id = childPipelineId;
+            pipelineConfig.configPipelineId = pipeline.id;
+
+            const childPipeline = new PipelineModel(pipelineConfig);
+
+            const configPipelineSecrets = [
+                {
+                    name: 'TEST',
+                    value: 'testvalue',
+                    allowInPR: true,
+                    pipelineId: pipeline.id
+                },
+                {
+                    name: 'ANOTHER',
+                    value: 'anothervalue',
+                    allowInPR: true,
+                    pipelineId: pipeline.id
+                }
+            ];
+            const childPipelineSecrets = [
+                {
+                    name: 'TEST',
+                    value: 'SHOULD OVERRIDE',
+                    allowInPR: true,
+                    pipelineId: pipeline.id
+                }
+            ];
+
+            const listConfig = {
+                params: {
+                    pipelineId: childPipeline.id
+                },
+                paginate
+            };
+
+            secretFactoryMock.list.resolves([
+                [childPipelineSecrets],
+                [configPipelineSecrets]
+            ]);
+            // when we fetch secrets it resolves to a promise
+            assert.isFunction(childPipeline.secrets.then);
```

Comment:
In the test, can we assert that the values we're getting is what we expect? not sure why other getter tests are structured like this, but can we try something like `return pipeline.secrets.then((secrets) => assert the value)`